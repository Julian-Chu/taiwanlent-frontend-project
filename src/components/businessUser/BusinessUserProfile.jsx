
import React, { Component } from 'react';
// import regions from '../common/regions';
import { connect } from 'react-redux';
import { Field, reduxForm, reset } from 'redux-form';
// import checkRules from '../../regularExpression/checkRules';
import { GetBusinessUserData, UpdateBusinessUserData } from '../../actions/businessuser';
import validate from './BusinessUserRegisterPages/validate';
import renderField from '../renderComponents/renderField';
import renderRadio from '../renderComponents/renderRadio';
import '../../styles/ReduxForm.css';


export class BusinessUserProfile extends Component {
  constructor(props) {
    super(props);
    this.state = {
      disabled: true
    };

    this.onFormSubmit = this.onFormSubmit.bind(this);
  }

  componentDidMount() {
    // this.props.setHeaderNontransparent();
    this.props.GetBusinessUserData();
  }

  onFormSubmit(values) {
    // console.log(values);
    // console.log('this.props:', this.props);
    // var history = this.props.history;
    // console.log('history: ', history);
    this.props.UpdateBusinessUserData(values, ()=>this.toggleChangeInput(true));
  }

  toggleChangeInput(disabled) {
    this.setState({
      disabled: disabled
    })
  }

  renderButtons(handleSubmit, pristine, submitting) {
    if (this.state.disabled) {
      return (
        <div>
          <button type="button" className="button button-border button-dark button-circle" onClick={() => this.toggleChangeInput(false)}>修改資料</button>
        </div>
      )
    } else {
      return (
        <div>
          {
            !(pristine || submitting) &&
            <button type="button" className="button button-border button-dark button-circle" onClick={handleSubmit(this.onFormSubmit)}>confirm</button>
          }
          <button type="button" className="button button-border button-dark button-circle" onClick={() => {
            this.props.reset('BusinessUserProfileForm');
            this.toggleChangeInput(true);
          }

          }>Cancel</button>

        </div>
      )
    }
  }



  render() {
    const { handleSubmit, pristine, submitting } = this.props;
    return (
      <div className="content-wrap">
        <div className="container clearfix">
          <div className="col_two_third col_last nobottommargin">
            <h1>廠商註冊</h1>
            <form id="register-form" name="register-form" className="nobottommargin" onSubmit={handleSubmit(this.onFormSubmit.bind(this))} disabled="true">
              {/* <Field
                name="username"
                placeholder=""
                className="col_half"
                title="使用者名稱"
                component={renderField}
                disabled="true"
              /> */}
              <div className="col_half">
                <label htmlFor="">使用者名稱</label>
                <div>{this.props.initialValues.username}</div>
              </div>
              <div id="gender">
                <label>性別</label>
                <div>
                  <Field
                    name="gender"
                    required={true}
                    options={[
                      { title: '男', value: 'male' },
                      { title: '女', value: 'female' }
                    ]}
                    component={renderRadio}
                    disabled={this.state.disabled}
                  ></Field>
                </div>
              </div>
              <div className="clear"></div>

              <Field
                name="email"
                placeholder=""
                className="col_half"
                title="Email"
                component={renderField}
                disabled={this.state.disabled}
              />
              {/* <Field
                name="reemail"
                placeholder=""
                className="col_half col_last"
                title="再次確認Email"
                component={renderField}
                disabled={this.state.disabled}
              /> */}
              <div className="clear"></div>
              {/* <Field
                name="password"
                placeholder=""
                className="col_half"
                title="輸入密碼"
                type="password"
                component={renderField}
                disabled={this.state.disabled}
              />
              <Field
                name="repassword"
                placeholder=""
                className="col_half col_last"
                title="再次輸入密碼"
                type="password"
                component={renderField}
                disabled={this.state.disabled}
              />
              <div className="clear"></div> */}

              <Field
                name="name"
                placeholder=""
                className="col_half"
                title="真實姓名"
                component={renderField}
                disabled={this.state.disabled}
              />
              <Field
                name="phone"
                placeholder=""
                className="col_half col_last"
                title="連絡電話"
                component={renderField}
                disabled={this.state.disabled}
              />
              <div className="clear"></div>
              <Field
                name="companyName"
                title="公司名稱/個人委託"
                placeholder=""
                className="col_half "
                disabled={this.state.disabled}
                component={renderField} />

              <Field
                name="department"
                title="部門"
                placeholder=""
                className="col_half col_last"
                disabled={this.state.disabled}
                component={renderField} />
              <div className="clear"></div>

              <Field
                name="companyLocation"
                title="公司所在城市"
                placeholder=""
                className="col_half "
                disabled={this.state.disabled}
                component={renderField} />

              <Field
                name="address"
                title="地址"
                placeholder=""
                className="col_half col_last"
                disabled={this.state.disabled}
                component={renderField} />
              <div className="clear"></div>

              <Field
                name="industry"
                title="產業類別"
                placeholder=""
                className="col_half"
                disabled={this.state.disabled}
                component={renderField} />

              <Field
                name="productIntroduction"
                title="產品簡介"
                placeholder=""
                className="col_half col_last"
                disabled={this.state.disabled}
                component={renderField} />
              <div className="clear"></div>
              <div className="col_half col_last">
                <br />
              </div>

              <div className="clear"></div>
              <div className="col_full nobottommargin">
                {this.renderButtons(handleSubmit, pristine, submitting)}
              </div>
            </form>
          </div>
        </div>
      </div>
    );
  }
}

BusinessUserProfile = reduxForm({
  validate,
  form: 'BusinessUserProfileForm',
  enableReinitialize: true
})(BusinessUserProfile);

const ConnectedBusinessUserProfile = connect(
  state => {
    var initialValues = state.businessUserData;
    return {
      initialValues
    }
  }, { GetBusinessUserData, UpdateBusinessUserData, reset })(BusinessUserProfile)


export default ConnectedBusinessUserProfile;

// export default reduxForm({
//   validate,
//   form: 'BusinessUserRegisterForm'
// })(connect(null, actions)(BusinessUserProfile));