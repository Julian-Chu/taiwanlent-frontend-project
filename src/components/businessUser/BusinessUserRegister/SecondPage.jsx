
import React from 'react';
import { Field, reduxForm } from 'redux-form';
import validate from './validate';
import renderField from './renderField';
import { connect } from 'react-redux';
import { fillUpUserData} from '../../../actions/businessuser';

const SecondPage = props => {
  const onFormSubmit = (values)=>{
    props.fillUpUserData(values, props.history);
  }
  const { handleSubmit, previousPage, pristine, submitting } = props;
  return (
    <form onSubmit={handleSubmit(onFormSubmit)}>
      <Field
        name="companyName"
        title="公司名稱/個人委託"
        placeholder=""
        className="col_half "
        component={renderField} />

      <Field
        name="department"
        title="部門"
        placeholder=""
        className="col_half col_last"
        component={renderField} />
      <div className="clear"></div>

      <Field
        name="companyLocation"
        title="公司所在城市"
        placeholder=""
        className="col_half "
        component={renderField} />

      <Field
        name="address"
        title="地址"
        placeholder=""
        className="col_half col_last"
        component={renderField} />
      <div className="clear"></div>

      <Field
        name="industry"
        title="產業類別"
        placeholder=""
        className="col_half"
        component={renderField} />

      <Field
        name="productIntroduction"
        title="產品簡介"
        placeholder=""
        className="col_half col_last"
        component={renderField} />

      <div className="clear"></div>
      <button type="button" onClick={previousPage} className="button button-border button-dark button-circle">Previous</button>
        <button type="submit" disabled={pristine || submitting} className="button button-border button-dark button-circle">Submit</button>
    </form>
  )
}
export default reduxForm({
  validate,
  form: 'BusinessUserRegisterForm',
  destroyOnUnmount: false,
  forceUnregisterOnUnmount: true,

})(connect(null, {fillUpUserData})(SecondPage));