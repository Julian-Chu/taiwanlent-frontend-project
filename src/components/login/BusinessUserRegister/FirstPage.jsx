import React from 'react';
import { Field, reduxForm } from 'redux-form';
import validate from './validate';
import renderField from './renderField';

const FirstPage = props => {
  const { handleSubmit } = props;
  return (
    <form onSubmit={handleSubmit}>
      <Field
        name="username"
        placeholder=""
        className="col_half"
        title="使用者名稱"
        component={renderField}
      />
      <div id="gender">
        <label>性別</label>
        <div>
          <label><Field name="gender" component="input" type="radio" value="male"></Field>男</label>
          <label><Field name="gender" component="input" type="radio" value="female"></Field>女</label>
        </div>
      </div>
      <div className="clear"></div>

      <Field
        name="email"
        placeholder=""
        className="col_half"
        title="Email"
        component={renderField}
      />
      <Field
        name="reemail"
        placeholder=""
        className="col_half col_last"
        title="再次確認Email"
        component={renderField}
      />
      <div className="clear"></div>
      <Field
        name="password"
        placeholder=""
        className="col_half"
        title="輸入密碼"
        type="password"
        component={renderField}
      />
      <Field
        name="repassword"
        placeholder=""
        className="col_half col_last"
        title="再次輸入密碼"
        type="password"
        component={renderField}
      />
      <div className="clear"></div>
      <button type="submit">Next</button>
      </form>
  )
}
export default reduxForm({
  validate,
  form: 'BusinessUserRegisterForm',
  destroyOnUnmount: false,
  forceUnregisterOnUnmount: true,

})(FirstPage);
// export default FirstPage;