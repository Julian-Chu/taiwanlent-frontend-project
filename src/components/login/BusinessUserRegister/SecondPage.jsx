
import React from 'react';
import { Field, reduxForm } from 'redux-form';
import validate from './validate';
import renderField from './renderField';

const SecondPage = props => {
  const { handleSubmit, previousPage } = props;
  return (
    <form onSubmit={handleSubmit}>

      <Field
        name="name"
        placeholder=""
        className="col_half"
        title="真實姓名"
        component={renderField}
      />
      <Field
        name="phone"
        placeholder=""
        className="col_half col_last"
        title="連絡電話"
        component={renderField}
      />
      <div className="clear"></div>
      <button type="button" onClick={previousPage}>Previous</button>
      <button type="submit">Next</button>
    </form>
  )
}
export default reduxForm({
  validate,
  form: 'BusinessUserRegisterForm',
  destroyOnUnmount: false,
  forceUnregisterOnUnmount: true,

})(SecondPage);