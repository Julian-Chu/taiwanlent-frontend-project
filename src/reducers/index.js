import { combineReducers} from 'redux';
import TalentsReducer from './reducer_talents';
import FilterReducer from './reducer_filters';

const rootReducer = combineReducers({
  talents: TalentsReducer,
  filter: FilterReducer
});

export default rootReducer;