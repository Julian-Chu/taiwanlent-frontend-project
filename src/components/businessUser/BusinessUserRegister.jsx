import React, { Component } from 'react';
import PropTypes from 'prop-types';
import FirstPage from './BusinessUserRegister/FirstPage';
import SecondPage from './BusinessUserRegister/SecondPage';

class BusinessUserRegister extends Component {
  constructor(props) {
    super(props);
    this.nextPage = this.nextPage.bind(this);
    this.previousPage = this.previousPage.bind(this);
    this.renderPageByNumber = this.renderPageByNumber.bind(this);
    this.state = {
      page: 1
    };
  }

  nextPage() {
    this.setState({ page: this.state.page + 1 });
  }
  previousPage() {
    this.setState({ page: this.state.page - 1 });
  }

  renderPageByNumber(page, onSubmit) {
    switch (page) {
      case 1:
        return (
          <FirstPage onSubmit={this.nextPage}></FirstPage>
        )
      case 2:
        return (
          <SecondPage onSubmit={onSubmit} previousPage={this.previousPage}></SecondPage>
        )
    }
  }

  render() {
    const { onSubmit } = this.props;
    const { page } = this.state;
    return (
      <div style={{margin: '5% 10% 5%'}}>
        <h1>廠商註冊</h1>
        {this.renderPageByNumber(page, onSubmit)}
        <ol className="nav">
          <li>
            <button  className='active' >1</button>
          </li>
          <li>
            <button  className={this.state.index >= 1 ? 'active' : ''} >2</button>
          </li>
        </ol><h6>{this.state.page}/2</h6>
      </div>
    )
  }
}

export default BusinessUserRegister;