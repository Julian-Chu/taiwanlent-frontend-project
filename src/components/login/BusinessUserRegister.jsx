import React, { Component } from 'react';
import PropTypes from 'prop-types';
import FirstPage from './BusinessUserRegister/FirstPage';
import SecondPage from './BusinessUserRegister/SecondPage';

class BusinessUserRegister extends Component {
  constructor(props) {
    super(props);
    this.nextPage = this.nextPage.bind(this);
    this.previousPage = this.previousPage.bind(this);
    this.renderPageByNumber = this.renderPageByNumber.bind(this);
    this.state = {
      page: 1
    };
  }

  nextPage() {
    this.setState({ page: this.state.page + 1 });
  }
  previousPage() {
    this.setState({ page: this.state.page - 1 });
  }

  renderPageByNumber(page) {
    switch (page) {
      case 1:
        return (
        <FirstPage onSubmit={this.nextPage}></FirstPage>          
      )
      case 2:
        return (
        <SecondPage onSubmit={this.nextPage} previousPage={this.previousPage}></SecondPage>
      )
      case 3:
        return <div>Page 3 </div>
    }
  }

  render() {
    const { onSubmit } = this.props;
    const { page } = this.state;
    return (
      <div>
        <h1>廠商註冊</h1>
        {this.renderPageByNumber(page)}
      </div>
    )
  }
}

export default BusinessUserRegister;