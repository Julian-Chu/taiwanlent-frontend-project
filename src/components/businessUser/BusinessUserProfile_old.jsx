import React, { Component } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { GetBusinessUserData } from '../../actions/businessuser';
import BusinessUserRegister from './BusinessUserRegister';

class BusinessUserProfile extends Component {
  constructor(props) {
    super(props);
    this.state = {
      modifyData: false
    }
  }
  componentDidMount() {
    this.props.GetBusinessUserData();
  }
  renderField(field) {
    return (
      <div>
        <label htmlFor="">{field.title}</label>
        <label htmlFor="">{field.value}</label>
      </div>
    )
  }

  renderUserData() {
    return (
      <div>
        <div className="col_half">
          <label>使用者名稱:</label>
          <p>{this.props.userdata.username}</p>
          <label>Email:</label>
          <p>{this.props.userdata.email}</p>
          <label>聯絡電話:</label>
          <p>{this.props.userdata.phone}</p>
          <label>部門名稱:</label>
          <p>{this.props.userdata.department}</p>
          <label>公司所在城市:</label>
          <p>{this.props.userdata.companyLocation}</p>
          <label>地址:</label>
          <p>{this.props.userdata.address}</p>
        </div>
        <div className="col_half col_last">
          <label>真實姓名:</label>
          <p>{this.props.userdata.name}</p>

          <label>性別:</label>
          <p>{this.props.userdata.gender}</p>
          <label>公司名稱/個人委託:</label>
          <p>{this.props.userdata.companyName}</p>
          <label>產業類別:</label>
          <p>{this.props.userdata.industry}</p>
          <label>產品簡介:</label>
          <p>{this.props.userdata.productIntroduction}</p>
        </div>
        <button onClick={() => this.setState({ modifyData: true })}>修改資料</button>

      </div>
    )
  }
  render() {
    return (
      <div className="container">
        <h1>用戶資料</h1>
        {this.state.modifyData? <BusinessUserRegister/>: this.renderUserData()}
      </div>

    )
  }
}

function mapStateToProps(state) {
  console.log(state);
  return { userdata: state.businessUserData }
};

// const mapDispatchToProps = dispatch =>(
//   {
//     GetBusinessUserData:()=>dispatch(businessUserActions.GetBusinessUserData())
//   }
// )
function mapDispatchToProps(dispatch) {
  return bindActionCreators({ GetBusinessUserData }, dispatch);
}


export default connect(mapStateToProps, mapDispatchToProps)(BusinessUserProfile);