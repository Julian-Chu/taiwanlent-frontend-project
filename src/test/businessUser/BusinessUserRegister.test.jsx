import React from 'react';
import { shallow, mount} from 'enzyme';
import {BusinessUserRegister} from '../../components/businessUser/BusinessUserRegister';
import { createStore, applyMiddleware} from 'redux';
import thunk from 'redux-thunk';
import reducers from '../../reducers';

describe('BusinessUserRegister',()=>{
  const store = createStore(reducers);
  const businessUserRegister = shallow(<BusinessUserRegister store={store}/>)

  it('check snapshot',()=>{
    expect(businessUserRegister).toMatchSnapshot();
  })
  it('initial state: page =1',()=>{
    expect(businessUserRegister.state().page).toEqual(1);
  })

  it('contains FirstPage initially',()=>{
    expect(businessUserRegister.find('Connect(FirstPage)').exists()).toBeTruthy();
  })


})