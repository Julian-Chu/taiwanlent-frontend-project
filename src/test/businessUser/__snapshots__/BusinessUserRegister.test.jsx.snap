// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BusinessUserRegister check snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BusinessUserRegister
    store={
        Object {
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
            Symbol(observable): [Function],
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          廠商註冊
</h1>,
        <ReduxForm
          onSubmit={[Function]}
/>,
        <ol
          className="nav"
/>,
        <h6>
          1
          /2
</h6>,
      ],
      "style": Object {
        "margin": "5% 10% 5%",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "廠商註冊",
        },
        "ref": null,
        "rendered": "廠商註冊",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "nav",
        },
        "ref": null,
        "rendered": null,
        "type": "ol",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            1,
            "/2",
          ],
        },
        "ref": null,
        "rendered": Array [
          1,
          "/2",
        ],
        "type": "h6",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            廠商註冊
</h1>,
          <ReduxForm
            onSubmit={[Function]}
/>,
          <ol
            className="nav"
/>,
          <h6>
            1
            /2
</h6>,
        ],
        "style": Object {
          "margin": "5% 10% 5%",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "廠商註冊",
          },
          "ref": null,
          "rendered": "廠商註冊",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "nav",
          },
          "ref": null,
          "rendered": null,
          "type": "ol",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              1,
              "/2",
            ],
          },
          "ref": null,
          "rendered": Array [
            1,
            "/2",
          ],
          "type": "h6",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
